### DISCUSSION TITLE
Can IGL be used to compute normal maps?

### DISCUSSION THREAD

**Discussion by r03ert0** [libigl/libigl#1775] [Q&A] [answered]
What could be a method to compute a normal map for a low poly mesh given a more detailed version of the mesh, so that normal map + low poly mesh approximate the detailed mesh?

**Comment by jdumas** [comment, marked as answer]
I don't think libigl has that feature yet. You can try to use a professional software such as [Substance Painter](https://docs.substance3d.com/bake/substance-bakers-172818436.html) or [Simplygon](https://documentation.simplygon.com/SimplygonSDK_8.3.31500.0/articles/starthere/featurelist/materiallod.html). The keyword you want to use is "baking" (normal map baking, material baking, etc.). Maybe Blender also has a similar feature, but if so I am not familiar with it.

**Reply by r03ert0** [reply]
thank you!

**Comment by alecjacobson** [comment]
If your mesh already has a good UV map then you can append the uv
coordinates to your xyz and run igl::decimate on the [xyzuv] 5d mesh to
create the low res mesh. Then the UV domain becomes the mapping used to
bake the details into a normal/displacement map.

The Seamless software based on libigl will do this even for meshes with
complex uv cuts https://github.com/songrun/SeamAwareDecimater

>
